### PR TITLE
OCM-2373 | feat: Moved helpers associated with getRole

### PR DESCRIPTION
**Pull request by den-rgb**
Associated with: https://github.com/openshift/rosa/pull/1547

# Jira

[OCM-2373](https://issues.redhat.com/browse/OCM-2373)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,7 +20,8 @@ require (
 	github.com/json-iterator/go v1.1.12 // indirect
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.2 // indirect
-	github.com/openshift-online/ocm-sdk-go v0.1.362 // indirect
+	github.com/openshift-online/ocm-sdk-go v0.1.370 // indirect
+	github.com/openshift/rosa v1.2.28
```
Comment (ciaranRoche) [inline go.mod:24]: We should not be importing ROSA CLI

File: pkg/aws/validations/iam_helpers.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	name := fmt.Sprintf("%s-%s-Role", prefix, role)
+	if len(name) > 64 {
+		name = name[0:64]
+	}
+	return name
+}
+
+func IsManagedRole(roleTags []*iam.Tag) bool {
+    for _, tag := range roleTags {
+        if aws.StringValue(tag.Key) == tags.ManagedPolicies && aws.StringValue(tag.Value) == "true" {
```
Comment (ciaranRoche) [inline pkg/aws/validations/iam_helpers.go:21]: `tags.ManagedPolicies` should be moved from ROSA to common

File: pkg/aws/validations/iam_helpers.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    return false
+}
+
+func HasCompatibleVersionTags(iamTags []*iam.Tag, version string) (bool, error) {
+	if len(iamTags) == 0 {
+		return false, nil
+	}
+	for _, tag := range iamTags {
+		if aws.StringValue(tag.Key) == tags.OpenShiftVersion {
```
Comment (ciaranRoche) [inline pkg/aws/validations/iam_helpers.go:34]: `tags.OpenShiftVersion` should also be moved from ROSA to common

File: pkg/aws/validations/iam_helpers.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/openshift/rosa/pkg/aws/tags"
+	semver "github.com/hashicorp/go-version"
+)
+
+func GetRoleName(prefix string, role string) string {
+	name := fmt.Sprintf("%s-%s-Role", prefix, role)
+	if len(name) > 64 {
```
Comment (ciaranRoche) [inline pkg/aws/validations/iam_helpers.go:13]: Can we save 64 as a const and export it and update it in ROSA, looks like we have it hard coded in places.

File: pkg/aws/validations/iam_helpers.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	}
+	for _, tag := range iamTags {
+		if aws.StringValue(tag.Key) == tags.OpenShiftVersion {
+			if version == aws.StringValue(tag.Value) {
+				return true, nil
+			}
+			wantedVersion, err := semver.NewVersion(version)
+			if err != nil {
+				return false, err
+			}
```
Comment (ciaranRoche) [inline pkg/aws/validations/iam_helpers.go:41]: Can we check this outside the loop. Is there a reason we are doing this ever iteration?

Comment (den-rgb) [reply pkg/aws/validations/iam_helpers.go]: I can move that logic to an external function but I don't see a way of checking it outside of the for loop without making it messy

Comment (den-rgb) [reply pkg/aws/validations/iam_helpers.go]: oh my mistake I thought you meant both current and wanted version

**Review by ciaranRoche — APPROVED** [review]

**Pull request merged**